File: 프로그래머스/Level1/약수의 개수와 덧셈.py
# https://programmers.co.kr/learn/courses/30/lessons/77884

def solution(left, right):
    def isWhat(n):
        cnt = 0
        for i in range(1, n+1):
            if n % i == 0:
                cnt += 1
        if cnt % 2 == 0:
            return True
        return False
    answer = 0
    for i in range(left, right + 1):
        if isWhat(i):
            answer += i
        else:
            answer -= i
    return answer

File: 프로그래머스/Level1/약수의 개수와 덧셈.js
// https://programmers.co.kr/learn/courses/30/lessons/77884

const count = (num) => {
	if (num === 1) return 1;
	let cnt = 2;
	for (let i = 2; i < num; i++) {
		if (num % i === 0) cnt++;
	}
	return cnt;
};

const solution = (left, right) => {
	let answer = 0;
	for (let i = left; i <= right; i++) {
		const temp = count(i);
		if (temp % 2 === 0) answer += i;
		else answer -= i;
	}
	return answer;
};
